source/common/upstream/upstream_impl.cc

```diff
@@ -865,7 +867,8 @@ void StrictDnsClusterImpl::ResolveTarget::startResolve() {
 
         HostVector hosts_added;
         HostVector hosts_removed;
-        if (parent_.updateDynamicHostList(new_hosts, hosts_, hosts_added, hosts_removed, false)) {
+        if (parent_.updateDynamicHostList(new_hosts, hosts_, hosts_added, hosts_removed, false,
```

Comment:
Hmm. I think the fact that this universally passed false previous was an artifact of old code before I went through and unified the logic for all cluster types. You will need to check me on this, but I think that this call should use exactly the same logic as the EDS code. If so, I actually wonder if these parameters can be removed somehow and the function can infer it for itself?

Reply:
That sounds reasonable to me - I think I already have access to both the `ClusterInfo` and `health_checker_` in `updateDynamicHostList`.

It'll mean that STRICT_DNS would also respect the active health check before draining connections when the flag introduced in this PR isn't set, but I take it that's desirable?